src/software/ai/hl/stp/play/free_kick_play.cpp

```diff
@@ -140,21 +147,9 @@ PassWithRating FreeKickPlay::shootOrFindPassStage(
     std::array<std::shared_ptr<CreaseDefenderTactic>, 2> crease_defender_tactics,
     const WorldPtr &world_ptr)
 {
-    auto pitch_division =
-        std::make_shared<const EighteenZonePitchDivision>(world_ptr->field());
-
-    PassGenerator<EighteenZoneId> pass_generator(pitch_division,
-                                                 ai_config.passing_config());
-
-    using Zones = std::unordered_set<EighteenZoneId>;
-
-    auto pass_eval = pass_generator.generatePassEvaluation(*world_ptr);
-    PassWithRating best_pass_and_score_so_far = pass_eval.getBestPassOnField();
-
-    auto ranked_zones = pass_eval.rankZonesForReceiving(
-        *world_ptr, best_pass_and_score_so_far.pass.receiverPoint());
-    Zones cherry_pick_region_1 = {ranked_zones[0]};
-    Zones cherry_pick_region_2 = {ranked_zones[1]};
+    PassWithRating pass_with_rating = pass_generator.getBestPass(*world_ptr);
+    std::vector<Point> best_receiving_positions =
+        receiver_position_generator.getBestReceivingPositions(*world_ptr, 2);
```

Comment:
2 seems like a magic number here

Reply:
I agree, I just kept the same logic as the old implementation here, #2953 should overwrite this.